### ISSUE TITLE
Monodevelop-5.10.2.49

### ISSUE THREAD

**Issue by ArsenShnurkov** [gentoo/dotnet#160]
Monodevelop is packaged not in gentoo way

https://gist.githubusercontent.com/ArsenShnurkov/087186792d8dd155c7c3/raw/bed2360aa4c4aea0ba583440f11a93809ac871c0/stdin

<pre>Unable to find version '6.0.8' of package 'Newtonsoft.Json'.
Unable to find version '2.6.4' of package 'NUnit.Runners'.
Unable to find version '5.2.2' of package 'Microsoft.AspNet.Mvc'.
Unable to find version '3.2.2' of package 'Microsoft.AspNet.Razor'.
Unable to find version '3.2.2' of package 'Microsoft.AspNet.WebPages'.
Unable to find version '1.0.0.0' of package 'Microsoft.Web.Infrastructure'.
Unable to find version '1.0.81' of package 'LibGit2Sharp.NativeBinaries'.</pre>

1) it dependes on numerous submodules which are restored from file
https://raw.githubusercontent.com/mono/monodevelop/master/.gitmodules
during make dist

This causes usage of git-r3 eclass instead of downloading sources archives (i.e. doesn't use gentoo CDN for source codes. If github will be unavailable, then there will be no acccess to source codes through gentoo mirrors)

2) it downloads untrusted binary packages in SRC_URI. For some of there there are trusted gentoo alternatives (for NUnit for example)

**Comment by Heather** [comment]
@ArsenShnurkov yes ideally monodevelop needs hard rework, specially newer versions support

**Comment by ArsenShnurkov** [comment]
I am reading make dist section of Makefile. It contains sytax which is unknown to me.

 http://www.gnu.org/software/make/manual/make.html#Pattern-Rules   
Why do one want to use rules like "%-operation: echo ..." ? "all-recursive", "clean-recursive", "install-recursive", "uninstall-recursive", "distcheck-recursive" - they all match to rule %-recursive, but i don't understood how that % rule works differently in this different cases

**Comment by ArsenShnurkov** [comment]
We need a local copy of monodevelop-addins repository by the same reasons as we need to have a local copy of nuget repository (Gentoo is source-based distribution and should have an ability to build without a connection to internet).

**Comment by ArsenShnurkov** [comment]
Microsoft.AspNet.Mvc from nuget repository (and it's dependencies) is under EULA. This is awful.

**Comment by ArsenShnurkov** [comment]
see also 
https://bugs.gentoo.org/show_bug.cgi?id=589554
about package LibGit2Sharp.NativeBinaries 
which is downloaded from
https://www.nuget.org/packages/LibGit2Sharp.NativeBinaries
and declared to be build with
https://github.com/libgit2/libgit2sharp.nativebinaries

**Comment by ArsenShnurkov** [comment]
Unable to find version '3.0.1' of package 'NUnit.Engine'.
https://github.com/nunit/nunit/issues/1708

**Comment by mpkh** [comment]
@ArsenShnurkov current version became broken?

**Comment by ArsenShnurkov** [comment]
no, it's work in progress - https://github.com/ArsenShnurkov/shnurise/blob/master/dev-util/monodevelop/monodevelop-6.1_p2016062206.ebuild

**Comment by ArsenShnurkov** [comment]
I am trying to build this:
https://github.com/ArsenShnurkov/shnurise/blob/bfadda939263885a192b76c630a3f319538e902d/dev-util/monodevelop/monodevelop-6.1_p2016062206.ebuild
and receive this log:
http://pastehtml.com/view/gnksq7k9x.html
my extraction of the error text:
<pre>
Project "/var/tmp/portage/dev-util/monodevelop-6.1_p2016062206/work/monodevelop-6.1/external/fsharpbinding/.paket/paket.targets" (DownloadPaket target(s)):
					Target DownloadPaket:
						Executing: mono --runtime=v4.0.30319 /var/tmp/portage/dev-util/monodevelop-6.1_p2016062206/work/monodevelop-6.1/external/fsharpbinding/.paket/paket.bootstrapper.exe
						Cannot open assembly '/var/tmp/portage/dev-util/monodevelop-6.1_p2016062206/work/monodevelop-6.1/external/fsharpbinding/.paket/paket.bootstrapper.exe': No such file or directory.
/var/tmp/portage/dev-util/monodevelop-6.1_p2016062206/work/monodevelop-6.1/external/fsharpbinding/.paket/paket.targets: error : Command 'mono --runtime=v4.0.30319 /var/tmp/portage/dev-util/monodevelop-6.1_p2016062206/work/monodevelop-6.1/external/fsharpbinding/.paket/paket.bootstrapper.exe' exited with code: 2.
</pre>
since I don't have a clue what F# is, I am asking an assistance of @mpkh  :)

**Comment by mpkh** [comment]
@ArsenShnurkov packet is that thing https://github.com/fsprojects/Paket
we used also to have fsharpbindings as separated package in this overlay
as far as I remember it's build system was not so good and stable
if there is such compiling flag you can obviously try to introduce use flag for F#

in addition I guess it will be reasonable to share this with MD team

**Comment by piedar** [comment]
Monodevelop needs a [patched libgit2sharp](https://github.com/libgit2/libgit2sharp/compare/master...mono:xs-6.0-v2) to build.  Unfortunately, the patch doesn't apply cleanly against upstream.

**Comment by ArsenShnurkov** [comment]
monodevelop-6.0.0.4761 requires 2 dependencies:
cmake, git
and call to
addwrite "/usr/share/.mono/keypairs"
somewhere in the process before call to nuget restore

**Comment by ArsenShnurkov** [comment]
monodevelop-6.0.0.4761 is unable to open .sln file from roslyn project (says "Load failed"):
<img src="https://user-images.githubusercontent.com/188456/27992574-857a1cd2-64a0-11e7-94ea-905f36239063.gif" />

**Comment by ArsenShnurkov** [comment]
2 days ago
<a href="https://github.com/mono/monodevelop/releases/tag/monodevelop-7.1.0.1276">monodevelop-7.1.0.1276</a>
was released...

**Comment by ArsenShnurkov** [comment]
monodevelop 6.0.0.4761 crashes, when I attempt to "watch" a variable during debug...

**Comment by piedar** [comment]
@ArsenShnurkov Same here with the debugger crashes.  I've almost given up on it because it's so unstable and none of the newer tagged releases have built successfully on my machine for quite some time.

**Comment by EinstokFair** [comment]
How monodevelop addins should be installed:
https://github.com/mono/mono-addins/wiki/Addin-Discovery-Basics
